### PR TITLE
Moved interpolation function from runtimeAnimation to animation

### PR DESCRIPTION
**Pull request by kcoley**
- Moved interpolation from runtimeAnimation to animation
- Added documentation

### CODE REVIEW

File: src/Animations/babylon.animation.ts [hunk 1]
```diff
@@ -1,8 +1,21 @@
 ﻿module BABYLON {
+    /**
+     * Represents the range of an animation.
+     */
     export class AnimationRange {
+        /**
+         * Initializes the range of an animation.
+         * @param {string} name - The name of the animation range.
```
Comment (deltakosh) [inline src/Animations/babylon.animation.ts]: Types must not be set (they are inferred)

File: src/Animations/babylon.animation.ts [hunk 2]
```diff
@@ -1,8 +1,21 @@
...
+     * Represents the range of an animation.
+     */
     export class AnimationRange {
+        /**
+         * Initializes the range of an animation.
+         * @param {string} name - The name of the animation range.
+         * @param {number} from - The starting frame of the animation.
+         * @param {number} to - The ending frame of the animation.
+         */
         constructor(public name: string, public from: number, public to: number) {
```
Comment (deltakosh) [inline src/Animations/babylon.animation.ts]: You can run gulp typedoc-check to do a typedoc check
Here you need to also comment parameters as they are public

File: src/Animations/babylon.animation.ts [hunk 3]
```diff
@@ -1,8 +1,21 @@
 ﻿module BABYLON {
+    /**
+     * Represents the range of an animation.
```
Comment (deltakosh) [inline src/Animations/babylon.animation.ts]: No period at end of sentences :)

File: src/Animations/babylon.animation.ts [hunk 4]
```diff
@@ -204,6 +341,11 @@
             return node.getScene().beginDirectAnimation(node, [animation], 0, totalFrame, (animation.loopMode === 1), 1.0, onAnimationEnd);
         }
 
+        /**
```
Comment (deltakosh) [inline src/Animations/babylon.animation.ts]: this block seems alone ;)

File: src/Animations/babylon.animation.ts (hunk 1 above)
Comment (kcoley) [reply src/Animations/babylon.animation.ts]: Fixed

File: src/Animations/babylon.animation.ts (hunk 2 above)
Comment (kcoley) [reply src/Animations/babylon.animation.ts]: Fixed

File: src/Animations/babylon.animation.ts (hunk 3 above)
Comment (kcoley) [reply src/Animations/babylon.animation.ts]: All periods removed from ends of sentences in comments :)

**Comment by kcoley** [comment]
@deltakosh @bghgary I'll make one more pass to make sure I comply with the commenting conventions http://doc.babylonjs.com/how_to/contribute_to_api

**Comment by kcoley** [comment]
@deltakosh @bghgary I added @ignore to the public underscore properties and methods

File: src/Animations/babylon.animation.ts [hunk 5]
```diff
@@ -12,37 +31,95 @@
      * Composed of a frame, and an action function
      */
     export class AnimationEvent {
+        /**
+         * Specifies if the animation event is done
+         */
         public isDone: boolean = false;
-        constructor(public frame: number, public action: () => void, public onlyOnce?: boolean) {
+
+        /**
```
Comment (deltakosh) [inline src/Animations/babylon.animation.ts]: Must be removed

Comment (kcoley) [reply src/Animations/babylon.animation.ts]: Removed

**Pull request merged**